x-pack/plugins/security_solution/public/detection_engine/rule_management/api/hooks/use_fetch_rule_by_id_query.ts

```diff
@@ -30,7 +34,7 @@ export const useFetchRuleByIdQuery = (id: string, options?: UseQueryOptions<Rule
     async ({ signal }) => {
       const response = await fetchRuleById({ signal, id });
 
-      return transformInput(response);
+      return transformInput(transformRuleResponseToRule(response));
```

Comment:
@maximpn, why did this transformation become necessary? If the response was working fine without it for many releases, it makes me believe there might just be a simple type mismatch, rather than a real need to transform the data.

Reply:
There are two types [Rule](https://github.com/elastic/kibana/blob/main/x-pack/plugins/security_solution/public/detection_engine/rule_management/logic/types.ts#L209) defined in UI and [RuleResponse](https://github.com/elastic/kibana/blob/main/x-pack/plugins/security_solution/common/api/detection_engine/model/rule_schema/rule_schemas.ts#L543) defined in the common folder. Rule management API endpoints return `RuleResponse` so this way [fetchRuleById](https://github.com/elastic/kibana/blob/main/x-pack/plugins/security_solution/public/detection_engine/rule_management/api/api.ts#L203) fetches `RuleResponse` considered as `Rule`.

`Rule` and `RuleResponse` have subtle discrepancies coming from the [baseSchema](https://github.com/elastic/kibana/blob/main/x-pack/plugins/security_solution/common/api/detection_engine/model/rule_schema/rule_schemas.ts#L93). `baseSchema` is used to build up API endpoint input validation schemas (you can find it in `rule_schemas.ts` below) and `RuleResponse` schema as well. This way `RuleResponse` automatically includes legacy types like

- `machine_learning_job_id` as `string`
- rule based `throttle`
- arbitrary `meta`

while it only needed for request schemas.

Instead of inflating the scope of this PR and avoid conflicts with adopting OpenAPI schemas I added `transformRuleResponseToRule` as a reasonable compromise.

Another solution would be to type `useUpdateRuleMutation` as returning `Rule` instead of `RuleResponse`. With rolling back changes for `fetchRuleById`'s return type it will help to omit the transformation.